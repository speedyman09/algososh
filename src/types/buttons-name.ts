export enum ButtonName {
  AddHead = "addHead",
  AddTail = "addTail",
  DeleteHead = "deleteHead",
  DeleteTail = "deleteTail",
  AddByIndex = "addByIndex",
  DeleteByIndex = "deleteByIndex",
}