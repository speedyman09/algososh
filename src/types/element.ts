import { ElementStates } from "./element-states";

export type TElement = {
value: string;
color: ElementStates 
}
export type TElementNumber = {
  value: number;
  color: ElementStates 
  }