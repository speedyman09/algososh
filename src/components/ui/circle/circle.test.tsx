import { Circle } from "./circle";
import renderer from "react-test-renderer";
import { ElementStates } from "../../../types/element-states";

describe("CircleComponent", () => {
  test("без буквы", () => {
    const tree = renderer.create(<Circle />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("с буквами", () => {
    const tree = renderer.create(<Circle letter="a" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("с head", () => {
    const tree = renderer.create(<Circle head="a" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("с react-элементом в head", () => {
    const tree = renderer.create(<Circle head={<Circle />} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("с tail", () => {
    const tree = renderer.create(<Circle tail="a" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("с react-элементом в tail", () => {
    const tree = renderer.create(<Circle tail={<Circle />} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("с index", () => {
    const tree = renderer.create(<Circle index={7} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("с пропом isSmall ===  true", () => {
    const tree = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("в состоянии default", () => {
    const tree = renderer.create(<Circle />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("в состоянии changing", () => {
    const tree = renderer.create(<Circle state={ElementStates.Changing} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("в состоянии modified", () => {
    const tree = renderer.create(<Circle state={ElementStates.Modified} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
