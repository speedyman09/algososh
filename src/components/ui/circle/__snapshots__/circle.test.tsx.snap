// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleComponent без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`CircleComponent в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   changing"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`CircleComponent в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`CircleComponent в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   modified"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`CircleComponent с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-cy="head"
  >
    a
  </div>
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`CircleComponent с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  >
    7
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-cy="tail"
  />
</div>
`;

exports[`CircleComponent с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="head"
      />
      <div
        className="circle   default"
        data-cy="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-cy="letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-cy="index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="tail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`CircleComponent с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="head"
      />
      <div
        className="circle   default"
        data-cy="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-cy="letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-cy="index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="tail"
      />
    </div>
  </div>
</div>
`;

exports[`CircleComponent с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-cy="tail"
  >
    a
  </div>
</div>
`;

exports[`CircleComponent с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    >
      a
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`CircleComponent с пропом isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle  small default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;
